core/include/traccc/clusterization/detail/sparse_ccl.hpp

```diff
@@ -24,12 +27,18 @@ namespace detail {
 /// @param L an equivalance table
 ///
 /// @return the root of @param e
-template <template <typename> class vector_t>
-TRACCC_HOST_DEVICE inline unsigned int find_root(
-    const vector_t<unsigned int>& L, const unsigned int& e) {
+TRACCC_HOST_DEVICE inline unsigned int find_root(const unsigned int* L,
+                                                 unsigned int e,
+                                                 unsigned int size) {
+
+    // Don't complain about an unused variable in optimised builds.
+    (void)size;
+
     unsigned int r = e;
+    assert(r < size);
```

Comment:
```suggestion
    assert(r < L.size());
```

Reply:
Moved the code to using `vecmem::device_vector<unsigned int>`...